Bio/SeqIO/Interfaces.py

```diff
@@ -94,11 +94,37 @@ class SequenceWriter:
     SequentialSequenceWriter class instead.
     """
 
-    def __init__(self, handle):
+    def __init__(self, target, mode="w"):
         """Create the writer object.
 
         Use the method write_file() to actually record your sequence records.
         """
+        if mode == "w":
+            try:
+                target.write("")
+            except TypeError:
+                # target was opened in binary mode
+                raise ValueError("File must be opened in text mode.") from None
+            except AttributeError:
+                # target is a path
+                handle = open(target, mode)
+            else:
+                handle = target
+        elif mode == "wb":
+            try:
+                target.write(b"")
+            except TypeError:
+                # target was opened in text mode
+                raise ValueError("File must be opened in binary mode.") from None
+            except AttributeError:
+                # target is a path
+                handle = open(target, mode)
+            else:
+                handle = target
+        else:
+            raise RuntimeError("Unknown mode '%s'" % mode)
```

Comment:
ValueError?

Reply:
To me this is a `RuntimeError`. The main (and probably only) purpose of this class is to be used by the other submodules in `Bio.SeqIO`. Then, if the mode is not correct, it means something is wrong in Biopython; it's not a mistake by the user.

Reply:
Well OK. I can see a case for either exception.